internal/server/http/handler/v1/vehicle/handler.go

```diff
@@ -52,6 +52,49 @@ func GetAllVehicles(svc service.VehicleService) fiber.Handler {
 	}
 }
 
+// @Summary		Get all vehicles by type
+// @Description	Get all vehicles by type
+// @Id				get-all-vehicles-by-type
+// @Tags			Vehicle
+// @Produce		json
+// @Success		200	{object}	entities.VehicleListResponse
+// @Failure		400	{object}	HTTPError
+// @Failure		401	{object}	HTTPError
+// @Failure		403	{object}	HTTPError
+// @Failure		404	{object}	HTTPError
+// @Failure		500	{object}	HTTPError
+// @Router			/v1/vehicle/type/{type} [get]
```

Comment:
Maybe not a new endpoint, what about having it in `/vehicle` with a query parameter like this?

`GET /v1/vehicle?type=<type>`